src/UI/Window.h

```diff
@@ -154,14 +153,17 @@ class cWindow
 	
 	/** Called on shift-clicking to distribute the stack into other areas; Modifies a_ItemStack as it is distributed!
 	if a_ShouldApply is true, the changes are written into the slots;
-	if a_ShouldApply is false, only a_ItemStack is modified to reflect the number of fits (for fit-testing purposes)
-	*/
-	void DistributeStack(cItem & a_ItemStack, cPlayer & a_Player, cSlotArea * a_ExcludeArea, bool a_ShouldApply);
+	if a_ShouldApply is false, only a_ItemStack is modified to reflect the number of fits (for fit-testing purposes) */
+	virtual void DistributeStack(cItem & a_ItemStack, int a_Slot, cPlayer & a_Player, cSlotArea * a_ClickedArea, bool a_ShouldApply) = 0;
+
+	/** Called from DistributeStack() to distribute the stack into a_AreasInOrder; Modifies a_ItemStack as it is distributed!
```

Comment:
It is used so that any code can test how much of the stack will be left after distributing (like in the example you linked). The crafting code uses this to prevent shift-click crafting of too many items.

By using this code we actually fixed a slight bug in vanilla - for example if you shift-click-craft torches, but there's space only for two of them in your inventory, vanilla will still craft them and you will lose those two extra torches that won't fit. We won't craft, so you don't lose items.

My question is whether this behavior has been changed in your changes (the comment suggests that `a_ShouldApply` is more or less ignored).